src/api/api.go

```diff
@@ -19,7 +19,7 @@ import (
 	"github.com/treenq/treenq/src/services/cdk"
 )
 
-func New(conf Config) (http.Handler, error) {
+func New(conf Config, progressBuf *domain.ProgressBuf) (http.Handler, error) {
```

Comment:
if you want to inject we can create it there instead of main, it makes it simpler to build the IMO